### ISSUE TITLE
Add option to delete all user-defined waypoints

### ISSUE THREAD

**Issue by geocachermgo** [cgeo/cgeo#8994]
I have a cache with lots (>100) of waypoints extracted from the personal note. Is there a way to delete all waypoints at once? It tried to delete the cache, but it stays (including the waypoints) in memory, so when I reload the cache the old waypoints are still there.

**Comment by MagpieFourtyTwo** [comment]
Yes, there is, but it's pretty tricky and needs SQLite manager. BTDTNT.  :Smirk:

To be serious: I would consider this a feature request. Had the same problem too, with a Mystery Multi: solved this one in the PN at gc.com, then downloaded the cache to c:geo, which immediately created about 150 WPs (because of erroneous extraction of WPs from falsely detected "coordinates"). This cache was simply undoable this way, so I for real used the SQL way.

Most regretfully even the tick "Prevent waypoints from note" does not help in this case, as a) the cache was not saved offline before and b) the tick will be reset on refresh anyway (which BTW already has a FR, but can not search for it right now).

**Comment by moving-bits** [comment]
> It tried to delete the cache, but it stays (including the waypoints) in memory, so when I reload the cache the old waypoints are still there.

Deleted caches are kept for a three day grace period and deleted from memory only after that.

**Comment by moving-bits** [comment]
I changed the issue title to "Add option to delete all user-defined waypoints" and set the "feature request" label.

**Comment by geocachermgo** [comment]
Maybe we can split the buttons:
First Row: Add | Delete All
Second Row: Add Position as Ref | Park
![IMG_20200912_155603](https://user-images.githubusercontent.com/43813636/92997100-e0526700-f510-11ea-9f0b-a30ad04a2648.jpg)

**Comment by geocachermgo** [comment]
I wanted to add the label "Prio Low" but didnt find it. Can only special users change labels?

**Comment by MagpieFourtyTwo** [comment]
> Most regretfully even the tick "Prevent waypoints from note" does not help in this case, as a) the cache was not saved offline before and b) the tick will be reset on refresh anyway (which BTW already has a FR, but can not search for it right now).

Found it: #8415

**Comment by moving-bits** [comment]
> I wanted to add the label "Prio Low" but didnt find it. Can only special users change labels?

Yes, AFAIK only members of the c:geo team can add/remove labels, only standard labels that come with the issue template can be set otherwise.

I've added the "Prio low" label.

**Comment by geocachermgo** [comment]
As a suggestion: Add possibilty to completely delete a cache before the grace period? The sequence would be  Copy note to clipboard, clear note, completely delete cache, reload, prevent generating waypoints, add note again.
That would be sufficient for me.

**Comment by MagpieFourtyTwo** [comment]
Sounds like one step of the manual ``Purge`` to me, which is on my wishlist for a while already ... functions of such a purge:
- delete caches marked for deletion (plus of course their logs, images, attributes etc. ...)
- look for orphaned entries and delete them, too (logs and/or images w/o cache etc.)
- check/repair db consistency
- rebuild indexes
- vacuum database
- tbc ...

But should be a separate issue, I would say ...

**Comment by Lineflyer** [comment]
> But should be a separate issue, I would say ...

Feel free to open such an issue.
IMHO we could enhance the "delete orphaned data" function we already have. Currently it deletes only orphaned files from caches no longer saved. It could at least also empty the (72 hour) cache cache.

**Comment by MagpieFourtyTwo** [comment]
Thought about it, but [while investigating for orphaned entries](https://github.com/cgeo/cgeo/issues/8632#issuecomment-692670078) I came to the conclusion to not to do so, cause c::geo's db appears to be in really good shape - even after extensive use. Thus the work for such a purge will most probably not pay out ...

**Comment by geocachermgo** [comment]
Tanks for implementation.
Remark from testing: Works a little strange if I have have "extract waypoint from note" active:
Everything is removed, if you switch off "extract" and on again, twaypoints from the note are there again.
If you manually delete a waypoint it is immediately recreated. If the note is edited the re extract happens.
Options: 
1) Additionally switch off "extract" and note that in the warning
 2) Keep behavior but note that immediate re extract is blocked but will happen later
3) Allow immeadiate re extract (analogous to manual delete) and mention that in the warning " If you don't want to re-extract you have to switch-off extract or clear note"
I would vote for 3)

**Comment by moving-bits** [comment]
> My preference would be 3) followed by 1) 2)

Created issue #9026 for easier tracking.
